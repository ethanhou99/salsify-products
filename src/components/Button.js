import React, { Component } from 'react';

class Button extends Component {
  handleClick = () => {
    this.props.onUpdate();
  };

  render() {
    return (
      <button 
        type="button" 
        className="button-style" 
        onClick={this.props.onUpdate}>Clear
      </button>
    )
  }
}

export default Button;