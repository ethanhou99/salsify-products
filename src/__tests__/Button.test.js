import React from 'react';
import {cleanup, fireEvent, render} from '@testing-library/react';
import Button from '../components/Button';

// automatically unmount and cleanup DOM after the test is finished.
afterEach(cleanup);

describe('Button test', () => {
  it('Button should render correctly', () => {
    const component = render(<Button />)
    expect(component.firstChild).toMatchSnapshot()
   })

   it('Function should be called after button click', () => {
    const onClick = jest.fn();
    const {getByText} = render(<Button onUpdate={onClick}/>);
    fireEvent.click(getByText('Clear'))
    expect(onClick).toHaveBeenCalled();
  });
});