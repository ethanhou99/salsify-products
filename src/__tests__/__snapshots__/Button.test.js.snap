// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button test Button should render correctly 1`] = `undefined`;

exports[`Button test renders the component 1`] = `undefined`;
